### DISCUSSION TITLE
presentation role of TabMenu component

### DISCUSSION THREAD

**Discussion by mpolo205** [primefaces/community#956] [PrimeNG] [answered]
Hi everyone,

there is another accessibility issue in my app, that corresponds the TabMenu component. Testing the app with Lighthouse reports the following issue:
'Values ​​assigned to "role=""" are not valid ARIA roles.'
Lighthouse also prints this code with the warning:
`<a role="presentation" pripple="" class="p-ripple p-element p-menuitem-link ng-star-inserted p-menuitem-link-active" ... >`

I am using ng-template inside of p-tabMenu. The html looks like this:
```
<p-tabMenu [model]="items" class="pearl-menu">
  <ng-template pTemplate="item" let-item>
    <div pTooltip="{{ item.label }}" tooltipPosition="bottom">
      <button
        [ngClass]="{
          'disabled-btn': item.disabled
        }">
        {{ item.id }}
      </button>
      <label>{{ item.label }}</label>
    </div>
  </ng-template>
</p-tabMenu>
```
My items object I am using implements a routerLink, that does not correspond with the role attribute 'presentation'
I am trying to overwrite the role
```
  ngAfterViewInit() {
    const newRole = document.querySelector('.p-menuitem-link');
    if (newRole) {
      newRole.setAttribute('role', 'button');
    }
  }
```
, but this does not work. Is there a way to solve that? Thank you very much.
I am using version 15.2.0.

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Please try this with v17. Accessibility support is not available in v15.
